WcaOnRails/app/models/registration.rb

```diff
@@ -263,6 +263,16 @@ def self.wcif_json_schema
     end
   end
 
+  validate :forcing_competitors_to_add_comment
+  private def forcing_competitors_to_add_comment
+    if competition&.force_comment_in_registration.blank?
+      return
+    end
+    if comments.strip.empty?
```

Comment:
now that I am looking at this with a fresher morning brain, I don't think you need both ifs anymore.
 ```
if competition&.force_comment_in_registration.present? && comments.strip.empty?
      errors.add(:user_id, I18n.t('registrations.errors.cannot_register_without_comment'))
    end
```

Reply:
Works too!